core/src/test/java/org/bitcoinj/wallet/authentication/AuthenticationGroupExtensionTest.java

```diff
@@ -148,4 +157,54 @@ public void roundTrip() throws UnreadableWalletException {
         AuthenticationKeyUsage platformUsage = mnext.getKeyUsage().get(platformKey);
         assertNull(null, platformUsage);
     }
+
+    @Test
+    public void keyTest() throws UnreadableWalletException {
+        DeterministicSeed seed = new DeterministicSeed(seedCode, null, passphrase, creationtime);
+        AuthenticationGroupExtension mnext = new AuthenticationGroupExtension(MainNetParams.get());
+        mnext.addKeyChains(MainNetParams.get(), seed, EnumSet.of(
+                AuthenticationKeyChain.KeyChainType.MASTERNODE_PLATFORM_OPERATOR,
+                AuthenticationKeyChain.KeyChainType.MASTERNODE_OWNER,
+                AuthenticationKeyChain.KeyChainType.MASTERNODE_VOTING,
+                AuthenticationKeyChain.KeyChainType.MASTERNODE_OPERATOR));
+        IDeterministicKey votingKey = mnext.getKeyChain(AuthenticationKeyChain.KeyChainType.MASTERNODE_VOTING).getKey(0);
+        IDeterministicKey ownerKey = mnext.getKeyChain(AuthenticationKeyChain.KeyChainType.MASTERNODE_OWNER).getKey(0);
+        IDeterministicKey operatorKey = mnext.getKeyChain(AuthenticationKeyChain.KeyChainType.MASTERNODE_OPERATOR).getKey(0);
+        IDeterministicKey platformKey = mnext.getKeyChain(AuthenticationKeyChain.KeyChainType.MASTERNODE_PLATFORM_OPERATOR).getKey(0, true);
```

Comment:
Here, the key derivation is handled by the `AuthenticationGroupExtension` object.